### DISCUSSION TITLE
Should reForge continue development, or be archived?

### DISCUSSION THREAD

**Discussion by Panchovix** [Panchovix/stable-diffusion-webui-reForge#109] [Announcements]
Hi there guys, hope you all are doing good!

I do this question, since now as we know, original Forge has being started to be updated, moving the backend from what it had before vs a completely new one https://github.com/lllyasviel/stable-diffusion-webui-forge.

This means, since he's the author he does know a lot of what he's doing, probably will implement new models architectures, etc. Also for sure he will update the built it in controlnet extension, while I don't have the knowledge for that.

For now the main differents/features that reForge has vs Forge, discarding the backend are:

* New samplers (Comfy samplers, Turbo samplers, CFG++ Samplers, ODE Samplers).
* Some custom schedulers (AYS Gits, 32 steps, etc).
* Skip early CFG, skip early cond, pad/unpad certain conditions on  Settings->Optimizations.
* Multiple checkpoints management (load multiple models, etc)
* Custom extensions ported from Comfy (RAUNet/MSW-MSA/AutomaticCFG/SkimmedCFG)
* Maybe I'm missing something more, but that comes to my mind.

So I ask, should reForge continue development? The thing is, I don't think me alone will be able to implement new models architectures into it, since I lack the knowledge to modify the patcher on forge_loader.py to manage comfy backend (the code to make new models is on dev_upstream, but since they aren't patched for forge/A1111, they don't work)

You can let me know what you guys think!

**Comment by Zotikus1001** [comment]
Maybe pausing adding new features is a good idea, and simply fix what needs fixing for now.
And lets see what the original creator does. Who knows, he dissapears again for who knows how long.

**Reply by Bullseye110** [reply]
For 5-6 months. :))

**Reply by ScionoicS** [reply]
disappointing to see people treat lllyasviel this way.  I think it would've been better if he kept the comfyui code as the backend and then the over all open software ecosystem would've been better off for it. Instead we're splintered now because comfydev lead the charge against using HIS code (that he borrowed from other projects) in forge.  

The culture of dogging on llly has to die and I think this project is a big keystone in keeping the hate sustained.  Something to think about @Panchovix

**Comment by user1x1user** [comment]
+1 - solidify the framework first, pretty toys later.

**Comment by altoiddealer** [comment]
I am also a project developer.  It is certain, I've dedicated no less than 1,500 hours to my project over the course of the past 14+ months.

My project has 31 stars (one is mine).  Sometimes I'll see a guy whip up a comfy node, post it, and it gets hundreds of stars all told in < 24 hours.

There's about 10 people I can confirm with certainty who have installed / used my project.  I'm sure there's more but there is zero buzz, no one reporting Issues, I'm mostly talking to the wall in my discord channel, I made a Reddit post with 0 replies, etc.

I'm working on it daily, and new features / improvements are always buzzing through my head.

**It's very rewarding to me.**  I wish it would get more attention, but it's not my motivation, I'm in love with the project.

---

It seems to me, that you put an S on your chest and saved the Forge community from abandonment with Forge.

All hope was lost, but you swooped in and saved us.  Personally, I'm very grateful for your efforts, and it's inspiring to see someone work so hard on such useful, and complicated software.

I don't know your motives, but my advice to you is just to do what makes you feel good.

I'm still running ReForge, but there is a very good chance I'll be switching over to main Forge again, depending on what's cooking there.

If your drive on this project is to fulfill the community's needs, everything is always changing quickly in this scene... time will only tell at this point.

**Reply by Bullseye110** [reply]
Where can we find this project?

**Reply by altoiddealer** [reply]
> Where can we find this project?

I didn't explicitly link to it as I wasn't trying to advertise it, per se - just wanted to make my feelings on the topic more tangible.
My project is a discord bot for TGWUI + ReForge / Forge / A1111
https://github.com/altoiddealer/ad_discordbot

**Comment by JinJomes** [comment]
OG Forge is headed to 'Experimental' - no guarantees of continuing benefit for low-end systems.

So far reForge (dev and Main) is faster than Forge for all projects I've imported over, just feels like there's a few bits here and there that need tightening-up/refining. Your rate of commits and additions is frankly impressive, if you wanted to stagger the roll-out of new features and instead build on the framework you've already put in place for the sake of stability/flow - there's already more than enough added features over Forge to keep us busy in the meantime!

**Reply by ScionoicS** [reply]
This post didn't age well since the first big experimental update was bringing all new memory management code to Forge and making it leaner. 

People speculating so hard with tinted lenses. The bias against llly is real here.

**Reply by altoiddealer** [reply]
> The bias against llly is real here.

Where is this coming from?  As far as I can tell, any comments made regarding Illy here are just common knowledge.

Unsure what rock you may have been living under, but damn near everyone in the SD community knows that within 1-3 weeks after Illy released Forge he simply vanished from the project development, allowing Issues and Pull Requests to pile up without comment, review, etc.  It was completely disheartening for contributors - myself included, I spent quite some time restoring the ControlNet Control Type API endpoint, only for my PR to just languish.  The project seemed essentially dead after 4-5 months of not a single peep.  The reason this project here (ReForge) exists at all is because @Panchovix took it into his own hands to preserve the legacy of Forge by forking it and continuing to update it.

Shortly after, Illy "returned from the dead" (in terms of Forge), probably spurred by insider knowledge of Flux model release.

There is definitely a chance of history repeating itself, where Illy considers the project status fine and poof, disappears again for months.

I would love to be using Forge right now, except the txt2img and img2img API endpoints are both bugged.

Everyone loves Illy, but it sure is nice when there is someone dedicated to stir the pot instead of a guy stirring dozens of pots meanwhile cooking all new dishes.

**Edit** Around the time I posted this comment, Illyasviel made the wonderful decision to give edit permissions to **Panchovix** on Forge main.  It's truly a beautiful day for Forge / ReForge, these guys together are definitely a powerhouse

**Reply by Bullseye110** [reply]
> > The bias against llly is real here.
> 
> Where is this coming from? As far as I can tell, any comments made regarding Illy here are just common knowledge.
> 
> Unsure what rock you may have been living under, but damn near everyone in the SD community knows that within 1-3 weeks after Illy released Forge he simply vanished from the project development, allowing Issues and Pull Requests to pile up without comment, review, etc. It was completely disheartening for contributors - myself included, I spent quite some time restoring the ControlNet Control Type API endpoint, only for my PR to just languish. The project seemed essentially dead after 4-5 months of not a single peep. The reason this project here (ReForge) exists at all is because @Panchovix took it into his own hands to preserve the legacy of Forge by forking it and continuing to update it.
> 
> Shortly after, Illy "returned from the dead" (in terms of Forge), probably spurred by insider knowledge of Flux model release.
> 
> There is definitely a chance of history repeating itself, where Illy considers the project status fine and poof, disappears again for months.
> 
> I would love to be using Forge right now, except the txt2img and img2img API endpoints are both bugged.
> 
> Everyone loves Illy, but it sure is nice when there is someone dedicated to stir the pot instead of a guy stirring dozens of pots meanwhile cooking all new dishes.

Exactly, man! But go to say this on the Forge Github page to see how much hate you'll get. I got hate only for pointing out what doesn't work very good after the gradio 4 update. But I really care about Forge as it was the best solution for the users without 4090s... The problem is that illyasviel just disappears from time to time for literally months and he didn't even listen the community to just create another branch for all of this "experimental" stuff he's now inserting into the main branch.
I respect him, but I'll point out when I find problems.

**Comment by jordanjay29** [comment]
In my mind, you've stabilized Forge and kept it viable for a daily driver. That's a vital cause. 

From an outsider's observations, the efforts of Forge's dev2 branch seemed to lead nowhere. If that had born more fruit, I would have suggested you consider refocusing to pull requests and contributions to the main repo. As it stands, I don't see an issue with an experimental and stable projects based on the same code, Forge can do the experiments it finds worthy, and if you feel it's worthy to continue ReForge, I think many would find the stability and reliability very comforting. 

(Your efforts saved me from going back to A1111, so I can't really recommend archiving this unless you're finished working on it. You have support from me.)

**Comment by metapea** [comment]
After i got `"HEAD is now at 29be1da7 pass options to cross attention class error: RPC failed; curl 28 Recu failure: Connection was reset fatal: expected 'acknowledgments' Press any key to continue"` when updating around the June 8 announcement my experience with OG Forge has been buggy, even when i found out that the extensions _sd-webui-cloud-inference_ or _stable-diffusion-webui-stable-horde_ was giving me `TypeError: 'NoneType' object is not iterable` more times then before, the LORAs stopped working out of nowhere. It's was only until i used reForge everything was fixed.
I say maybe be like [ytdl-patched](https://github.com/ytdl-patched/ytdl-patched), as a fork doing it own thing but with goal merging back into the OG git (maybe when you want to stop working on reforge). You should be a part of the OG forge team in my opinion. But i still prefer your coding on reForge over lllyasviel's forge so far since this is what the dev2 branch on OG forge was trying to do, now pulled off.
_Also i'm still waiting for a fix on #33_, i don't really care about gradio 4 when the backend for generating images still need to be worked on.

**Comment by Vinzero-SS** [comment]
I would still continue development, mostly because I don't trust lllyasviel to simply add or fix a few things for a month (or less) then abandon the project all over again, there's already a precedent with that dev.

There's also the fact lllyasviel stated that all future changes would be experimental and most likely break things, so I would not switch back to the main branch anytime soon.

Honestly, please consider continuing. ReForged has been nothing short of amazing.

**Comment by pinea00** [comment]
,I found that gradient 4 Forge deleted the entire ldm_patched directory, which will result in many extensions that specifically support forge being unusable in the future.（ic-light.deepcache.....）,If terminate reforge, you can only use these useful extensions on the old version.

**Comment by itsagoodbrain** [comment]
I think the experimental side of Forge is going to keep it from being daily use for the vast majority of the community. I really appreciate all you've done to update and revitalize the original parts that made Forge great. I think reForge is going to be what I recommend to people for a long time so I hope you'll keep it around and update it. Just don't burn yourself out worrying about every request that comes in. Keeping some updates rolling along will keep most of us around for a long time to come.

**Comment by toyssamurai** [comment]
What if the goal changes? As far as I can tell, SDForge won't be the same SDForge. It will be experimental, meaning if you need a stable solution for work, it won't be a suitable choice. So, how about aiming ReForge as a stable version of SDForge ... only includes features that deem stable for daily work. It may also include features that will never be added to SDForge because they do not align with its project goal.

**Comment by ICodeDiffusion** [comment]
Stable ReForge would be perfect and have little bits of pieces of new stuff as well every now and then.
It would be nice to have both Fore & Reforge since Forge is focusing on experimental, so it would be nice to fall back to this just in case. Not to mention checking out all the new samplers and schedulers you added.
If I were you I'd keep the project. It's really impressive what you have done :D, I honestly had no idea there were that many more samplers and schedulers haha and thank you for your amazing work! Keep it up!
Oh yeah and I really really didn't want to go back to A1111, it's still not close to being as fast.

**Comment by Lesteriax** [comment]
I'm extremely grateful for the work you've done so far. The speeds I'm getting vs auto1111 is vastly better and as others pointed out, adjusting current framework would be better than adding new features, unless they're super important.

**Comment by iwr-redmond** [comment]
![exemplar](https://private-user-images.githubusercontent.com/9307310/302143114-78c04e06-8ef0-4224-9c25-8f1bee9861de.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5MjczNzEsIm5iZiI6MTcyMjkyNzA3MSwicGF0aCI6Ii85MzA3MzEwLzMwMjE0MzExNC03OGMwNGUwNi04ZWYwLTQyMjQtOWMyNS04ZjFiZWU5ODYxZGUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDZUMDY1MTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTFiNmUyMGY0MzUwMDMyOTg0NTE5OWQxMWVjMjlkMTUxYzNjNmRjNmJlYTUzYjdlYmQxODIxOTM0ODYxN2MyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wo-aR94wvsBUUpRTf33LqVg6ZO--rFHzxWYY6E3BaDo)

Be the @mashb1t of Forge. Agree with view that stabilizing existing stuff is best.

**Comment by DanPli** [comment]
You said it yourself, Forge is changing completely. I don't think you should worry about implementing Forge's new vision at all. If people want to go experimental, they will install Forge. Meanwhile I am enjoying and loving reForge for what it is, the good old Forge but with more and new stuff added.

**Comment by zhaol4v** [comment]
if Illyas adds the samplers like CFG++ etc then I'll use his repo, until then I'm sticking with reforge

**Comment by StylusEcho** [comment]
With development resumed on the repo again, it doesn't change what his goal is for Forge, and neither does it change your own.

**Comment by Panchovix** [comment]
Thanks guys, I understand now. I will keep working on reForge, try to fix bugs that exist and take it slowly.

I will for sure try to implement new models and update controlnet, but I can't promise that these will be here with my knowledge.

**Reply by Bullseye110** [reply]
But you don't have to be alone in this, maybe you can make people like @mashb1t part of your project to help you with things you don't know for now.

**Comment by VcCart** [comment]
To increase the number of regular project participants, it would be nice to see a portable version of Reforge that beginners can download.
There would be more users who don't need to understand Git commands. After all, most artists and designers, at the beginning of the journey, simply do not know how it works...
And developers and programmers would have more statistical data to make decisions.
And in (issues) there would be more ideas and feedback.

**Reply by Panchovix** [reply]
I would love this, but I'm not sure how to do that at the moment. Any help for a pre done package is very welcome.

Someone posted this https://github.com/Panchovix/stable-diffusion-webui-reForge/discussions/85

But not sure if it's auto maintainable?

**Reply by iwr-redmond** [reply]
https://github.com/mklasen/zip-creation-github-action/blob/master/.github/workflows/release-action.yml

Seems possible

**Reply by metapea** [reply]
Alternatively, you can download the previous version of forge then update to reForge with git:
https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/849

**Comment by Bullseye110** [comment]
The best option would be to continue developing this project as its own, because we can't trust illyasviel that much for constant updates/fixes for Forge (he's good at disappearing for like 5-6 months, then he returns and so on - guess everyone knows that already). He's good at what his doing, but he abandons the project when you least expect it. Maybe you can get involved into the main Forge repo, talk with him.
As for ReForge, keep it as it is for now and only implement things from the "experimental" Forge project that are stable and working without broking the entire WebUI and/or the other extensions if you really wish to get new things from that repo.
For things you don't know, just write on github/reddit/discord and at least one person should answer. Maybe you will even make a small team with people like you. But you have to search (no one will come if you don't say that you need help - it's like calling 911, the ambulance won't come to help you if you don't call them). :)))
Good luck and keep up the good work! Hopefully you'll find some good members for the ReForge team.

Unfortunately I'm no coder! :(

**Comment by Paharven-Salin** [comment]
Dear Panchovix

Absolutely DO keep going please! You have done a great job so far and I am super grateful for not having to devolve to A1111 or learn to thread the comfy spagetthi tree, ugh.
Having a STABLE interface with full up-to-the-minute (or very close) support from the maintainer is fantastic. As others have said, Forge itself is in flux and not yet whole. It may be utterly awesome 3 months from now or a leaky ship with nobody at the wheel - hopefully the former but things and people are unpredictable.
You obviously have some decent knowledge and that will also evolve over time, especially with some help from other generous souls along the way.

Thanks for the continued work. Sadly I cannot offer any help as I am a mere user but you have my gratitude and that of many others I am sure.

Please don't get jaded by entitled demands from ill-mannered folk, focus on the joyful aspects and fix the big ticket items as you have time.

I wish you very well

**Comment by BNP1111** [comment]
First of all thank you for your contribution to keep running forge for those who have to run it for hardware reasons. Again, please follow your heart, when you're busy or feel annoyed, ignore this place and enjoy your life. We wish you and your project (from now on it's your project and not a "fork") the best of luck!

**Comment by daPhie79** [comment]
I just wanted to chime in and say I'm now using your fork exclusively, it combines so many of the best features all together! So I'm all for you continuing the project!

**Comment by TheVertexDoctor** [comment]
I say stick with it, the update to Forge has broken some addons I rely on such as the Civit++ browser. Until such time as the addons I rely on work with the new forge I am sticking with reForge.

**Comment by amitsandhel** [comment]
@Panchovix I suggest just wait a bit seems to be until August 11, until Forge is done adding their changes and then port the changes over to here reforge. 

At the end of the day it was stated that forge is experimental and so it will have breaking changes.

We do need a stable forge that is stable and usable and it's not like you can't take the features from forge and implement them here in reforge just to be safe. 

The best is to continue development of reforge in the case it does get abandoned again or the author of Forge gives you or a group of developers authority to his dev and main branches but that is unlikely

**Reply by altoiddealer** [reply]
> Just port the changes over

it is 1000% not that easy.  The original Forge repo was 5 months stagnant, then now Illyasviel returned to throw all that code into a blender.

What we have here in ReForge, is that old code but with 5 months of A1111 commits merged in with revisions to make them work in Forge.

I’m not saying it’s impossible but let’s not assume Panchovix has any intention of changing the scope of their project, or attempting to solve that puzzle.

**Reply by Panchovix** [reply]
I probably plan to add the features of reForge (samplers, schedulers, extensions, etc) into OG Forge new backend.

Porting the backend of Forge into reForge as it is now, well it's possible but it will basically break all extensions. So I will have separate branches for each.

**Reply by Bullseye110** [reply]
> I probably plan to add the features of reForge (samplers, schedulers, extensions, etc) into OG Forge new backend.
> 
> Porting the backend of Forge into reForge as it is now, well it's possible but it will basically break all extensions. So I will have separate branches for each.

What do you think about creating a Discord server for this project? It's easier for people to ask for help/show issues/errors and to share the knowledge.
If you create a server, write the invitation link somewhere. Or if you do not wan't to make a server from scratch, I have a server with 250 members around Stable Diffusion and I can make you an owner there.

**Reply by Panchovix** [reply]
I don't have a discord server, not sure if I can manage a lot of issues/errors at the same time outside github. Any help is welcome though, and I can join that server if you want to as well.

**Reply by Bullseye110** [reply]
> I don't have a discord server, not sure if I can manage a lot of issues/errors at the same time outside github. Any help is welcome though, and I can join that server if you want to as well.

https://discord.gg/XQ2CHEaGaB

I'll find 1 more moderator, so that we'll rearrange everything.

Everyone from here can join, I'll share methods for the people with weak GPUs to generate images with ReForge and many more interesting things.

**Comment by digby-dart** [comment]
I signed up to github to thank you for all your effort. Reforge keeps on getting more impressive with every update. Just like me, I imagine there must be many other creatives quietly enjoying the fruits of your coding, comfortably generating images and ideas on their minimalist hardware and budget. Enjoy your week.

**Comment by ghost** [comment]
The current forge has broken extensions such as regional prompter which I use heavily. Yes, if possible keep it up. We need alternatives. And if possible a master branch that isn't a WIP and we can rely on to run AI on!

**Comment by mweldon** [comment]
As others have said, Forge is going strong right now, but who knows for how long. ReForge is really solid. You've done an outstanding job with it. I don't think you need to archive it unless you just aren't feeling it anymore. I still use ReForge and will continue to do so, at least until I see a compelling reason to switch, though I also use Comfy for Flux and Invoke for inpainting and rescaling. But if I need to blast out a bunch of images or iterate on some prompts, then ReForge is my go-to.

**Comment by Paharven-Salin** [comment]
To add: certain FORGE features have issues still, adetailer for instance will not use "separate checkpoint" currently so having another solid version - REFORGE - is a massive thing. Reliable tools to get the job done are what we need and you are providing an excellent one. :)

**Comment by Panchovix** [comment]
As some of you guys have noticed, now I have been assigned as collaborator to the original Forge branch.

This means that the new-main-forge branch is now deprecated.

The other branches (main, dev_upstream) will be keep getting updated as A1111/Comfy gets updated (respectively).
dev_upstream_experimental is basically a playground where I will try to add some other features, including new models when I have the time, which maybe will be a bit hard now. For now, it works fine though, if you want to test some speeds optimizations.

Those branches probably will be updated slower, but in mind to keep them stable and compability with older extensions.

Any question is welcome!

**Reply by altoiddealer** [reply]
That's amazing news!

Linking it here as I just found the relevant post from Illyasviel...

https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1297

**Reply by ScionoicS** [reply]
Strong move

**Comment by ArnorWing** [comment]
reForge is the best! 
I really do hope development continues. 
I only use SDXL so reForge is perfect.

**Comment by VeteranXT** [comment]
Add Zluda support for AMD?

**Comment by Gushousekai195** [comment]
At this point, I really think this should continue development. I'd really like for there to be a way to implement Flux support on this one.

lllyasviel is currently AWOL from Forge development with no sign of return.

**Comment by Dfafb** [comment]
Dear author support the project as long as you can.
Personally, I have long tried to achieve on Forge style as on ComfyUI in the style of the eyes of characters, but did not get any way, and today put reForge and it is almost as in ComfyUI. I'm thrilled!
Thank you for your hard work!
